### PR TITLE
Add Windows-related ClusterClass Template changes

### PR DESCRIPTION
**Pull request by tenczar**
### What this PR does / why we need it
This PR adds a tkg-worker-windows class to the tkg-vsphere-default clusterclass. This worker class facilitates windows workload cluster creation. It is very limited in functionality, in particular, it does not properly support http proxies, nor does it support custom image repositories. It is worth discussing if supporting those should be in scope at the moment.

### Which issue(s) this PR fixes

Fixes #

### Describe testing done for PR
Verify that diffs correctly set Windows attributes where necessary. 

### Release note

```release-note

```

### PR Checklist

- [ ] Squash the commits into one or a small number of logical commits
      
- [ ] Use good commit [messages](https://github.com/vmware-tanzu/tanzu-framework/blob/main/CONTRIBUTING.md)
- [ ] Ensure PR contains terms all contributors can understand and links all contributors can access

### CODE REVIEW

File: pkg/v1/providers/infrastructure-vsphere/v1.1.0/cconly/overlay.yaml [hunk 1]
```diff
@@ -56,5 +54,26 @@ spec:
             name: tkg-vsphere-worker
             namespace: #@ data.values.NAMESPACE
 
-#@ end
+    #@overlay/match by=overlay.index(1)
```
Comment (saji-pivotal) [inline pkg/v1/providers/infrastructure-vsphere/v1.1.0/cconly/overlay.yaml:57]: Is there a better match condition that we can use?

File: pkg/v1/providers/infrastructure-vsphere/v1.1.0/yttcc/overlay.yaml [hunk 2]
```diff
@@ -73,6 +72,29 @@ spec:
         failureDomain: #@ data.values.VSPHERE_AZ_2
         #@ end
       #@ end
+      #@ else:
+      #@overlay/match by=overlay.index(0)
```
Comment (saji-pivotal) [inline pkg/v1/providers/infrastructure-vsphere/v1.1.0/yttcc/overlay.yaml:76]: Is there a better match criteria we can use instead of referencing an index?

**Review by saji-pivotal — CHANGES_REQUESTED** [review]
Looks good overall, but I have a couple questions

**Comment by anujc25** [comment]
Also please rebase with latest `package-based-lcm` branch. Any changes in `cconly` we need to run `./packages/management/tkg-clusterclass/hack/sync-cc.sh` to sync with packages.

File: pkg/v1/providers/infrastructure-vsphere/v1.1.0/yttcc/overlay.yaml (hunk 2 above)
Comment (tenczar) [reply pkg/v1/providers/infrastructure-vsphere/v1.1.0/yttcc/overlay.yaml:76]: This one we actually want to make overwrite the 0th element.

**Comment by vuil** [comment]
given there are some remaining caveats and has user-facing effects, it would be good to create an issue, mention the caveats, and point the folks who have contributed to the windows overlays to them.

(and mention said issue here. with Partially-fixes: #NNNN)

**Review by vuil — APPROVED** [review]
lgtm, thanks for the updates

**Review by vuil — APPROVED** [review]

**Review by saji-pivotal — APPROVED** [review]

**Pull request merged**